bonfire/Modules/Settings/Controllers/GeneralSettingsController.php

```diff
@@ -76,6 +76,31 @@ public function saveGeneral()
         setting('App.dateFormat', $this->request->getPost('dateFormat'));
         setting('App.timeFormat', $this->request->getPost('timeFormat'));
 
+        // Check for an logo to upload
+        if ($file = $this->request->getFile('siteLogo')) {
+
+            if ($file->isValid()) {
+
+                $filename = 'custom.'. $file->getExtension();
+                $logo_path = WRITEPATH .'uploads/logo';
+
+                if ($file->move($logo_path, $filename, true)) {
+
+                          service('image')
+                          ->withFile($logo_path .'/'.$filename)
+                          ->fit(150, 150, 'center')
+                          ->save($logo_path .'/med_'.$filename);
+
+                          service('image')
+                          ->withFile($logo_path .'/'.$filename)
+                          ->fit(30, 30, 'center')
+                          ->save($logo_path .'/thumb_'.$filename);
+
+                    setting('App.siteLogo', $filename);
+                }
+            }
```

Comment:
You forgot to handle the case when the file is not valid. The user should be notified in some way.